### PR TITLE
Merge maint 1.8

### PR DESCRIPTION
**Pull request by sgillies**

### CODE REVIEW

**Comment by sgillies** [comment]
@rbuffat I've done it! The merge-maint-1.8 branch has commits from maint-1.8 and resolves conflicts. Unless anything strikes you as wrong, I'll merge this later today and maint-1.9 will be up to date.

File: docs/manual.rst [hunk 1]
```diff
@@ -1007,7 +1004,7 @@ a schema of ``{'properties': {'bar': 'int', 'foo': 'str'}}`` will produce
 a shapefile where the first field is 'foo' and the second field is 'bar'. If
 you want 'bar' to be the first field, you must use a list of property items
 
-.. sourcecode:: python
+.. sourcecode:: pycon
```
Comment (snowman2) [inline docs/manual.rst:1007]: Is that correct?

Comment (sgillies) [reply docs/manual.rst:1007]: Good catch!

File: docs/manual.rst [hunk 2]
```diff
@@ -271,10 +270,6 @@ collection to get back to the beginning.
 Collection indexing
 -------------------
 
-.. admonition::
-
-   New in version 1.1.6
-
 Features of a collection may also be accessed by index.
 
 .. code-block:: pycon
```
Comment (snowman2) [inline docs/manual.rst:275]: Another one here.

File: docs/manual.rst [hunk 3]
```diff
@@ -303,13 +298,15 @@ Features of a collection may also be accessed by index.
...
 -------------
 
 A :py:class:`~fiona.collection.Collection` involves external resources. There's
 no guarantee that these will be released unless you explicitly
 :py:meth:`~fiona.collection.Collection.close` the object or use
-a :py:keyword:`with` statement. When a :py:class:`~fiona.collection.Collection`
+a :keyword:`with` statement. When a :py:class:`~fiona.collection.Collection`
 is a context guard, it is closed no matter what happens within the block.
 
 .. sourcecode:: pycon
```
Comment (snowman2) [inline docs/manual.rst:312]: Another one here

File: docs/manual.rst [hunk 4]
```diff
@@ -1157,7 +1155,7 @@ indexes.
     3
 
 To filter features by property values, use Python's builtin :py:func:`filter` and
-:py:keyword:`lambda` or your own filter function that takes a single feature
+:keyword:`lambda` or your own filter function that takes a single feature
 record and returns ``True`` or ``False``.
 
 .. sourcecode:: pycon
```
Comment (snowman2) [inline docs/manual.rst:1161]: Another one here

File: docs/manual.rst (hunk 2 above)
Comment (sgillies) [reply docs/manual.rst:275]: This one looks okay to me. The following code block does have python prompts and "pycon" is intended.

We have inconsistent use of code-block and sourcecode directives, but that's out of scope.

Comment (snowman2) [reply docs/manual.rst:275]: Sounds good, thanks for getting this MR ready :+1:

**Comment by rbuffat** [comment]
@snowman2 @sgillies 

The following code block is not valid Python and rstcheck returns this as an error. I assumed that `.. sourcecode:: pycon` results in the same formatting but without the code checking. Maybe it is better to [ignore python code checks ](https://github.com/myint/rstcheck/blob/master/README.rst#ignore-specific-languages) or do not use rstcheck on travis if it is annoying. 

```
c = fiona.open(
    '/tmp/file.shp',
    'w',
    schema={'properties': [('bar', 'int'), ('foo', 'str')], ...},
    ... )
```

**Comment by sgillies** [comment]
@rbuffat I fixed the example code. Thanks for the review, both!

**Comment by rbuffat** [comment]
@sgillies Thank you for all the work for merging this! 

Before I only had time to look at the rstcheck issues. I now had a glance at all the changes, the only thing I found so far is that Python 2.7 and GDAL < 2.4 was eliminated from travis but not from appveyor. Should then also the _shim modules be removed?

**Comment by sgillies** [comment]
@rbuffat it was a bad time :sweat_smile: I am determined to never let our branches become strangers again.

No, we won't drop the shims for 1.9. That's for 2.0. But I think it is okay to test compatibility with GDAL versions 1.11-2.3 less thoroughly than we have done.

**Pull request merged**